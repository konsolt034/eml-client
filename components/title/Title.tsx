"use client";

import styled from "styled-components";

type TitleProps = {
  color?: string;
  size?: string;
  margin?: string;
};

const Title = styled.h1<TitleProps>`
  color: ${(props) => props.color || "#f9f9f9"};
  font-size: ${(props) => props.size || "16px"};
  ${(props) => props.margin && `margin: ${props.margin}`}
`;

export default Title;
