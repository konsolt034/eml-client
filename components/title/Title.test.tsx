import { render } from "@testing-library/react";

import Title from "./Title";

describe("Component - Title", () => {
  it("renders title component", () => {
    const { container } = render(<Title>{"Gift a friend this holiday"}</Title>);
    expect(container).toMatchSnapshot();
  });

  it("should render a heading component", () => {
    const { getByRole } = render(<Title>{"Gift a friend this holiday"}</Title>);
    expect(getByRole("heading")).toBeTruthy;
  });
});
