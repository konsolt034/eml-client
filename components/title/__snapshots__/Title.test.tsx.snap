// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Component - Title renders title component 1`] = `
<div>
  <h1
    class="sc-bcPKhP jKfhls"
  >
    Gift a friend this holiday
  </h1>
</div>
`;
